giraph-core/src/main/java/org/apache/giraph/job/DefaultJobProgressTrackerService.java

```diff
@@ -129,6 +137,20 @@ public void run() {
   }
 
   /**
+   * Determine if the job will finish in allotted time
+   * @param conf Giraph configuration
+   * @param job Job
+   * @param workerCount Worker count
```

Comment:
Do you need workerCount here, should you use workersInSuperstep? If you need it, rename to startedWorkerCount or something, this way it sounds like how many workers job needs.

Reply:
this is not the started worker count, but all the worker counts that are requested by the job. My implementation in Digraph will only calculate the expected completion time if all workers are included in the CombinedWorkerCount to make the calculation accurate .